File: AnalysisStep/test/classlib/src/EventStream.cpp
#include <ZZAnalysis/AnalysisStep/test/classlib/include/EventStream.h>

void EventStream::AddEventSource(TString path, const std::function<bool(Tree*)>& cut)
{
    paths.push_back(path);
    cuts.push_back(cut);
}

std::vector<TString> EventStream::GetPaths()
{
    return paths;
}

std::vector<std::function<bool(Tree*)>> EventStream::GetCuts()
{
    return cuts;
}

File: AnalysisStep/test/Python/trainlib_legacy/EventStream.py
class EventStream:

    def __init__(self):
        self.paths = []
        self.cuts = []

    def AddEventSource(self, path, cut):
        self.paths.append(path)
        self.cuts.append(cut)

    def GetFiles(self):
        return self.paths

    def GetCuts(self):
        return self.cuts
